mava/components/training/model_updating.py

```diff
@@ -95,17 +99,31 @@ def on_training_utility_fns(self, trainer: SystemTrainer) -> None:
             None.
         """
 
-        # Initilaise running statisitics here
-
+        # Initilaise target values running statisitics here
         if self.config.normalize_target_values:
-            trainer.store.running_stats_fn = compute_running_mean_var_count
-            running_stats = jnp.array([0, 0, 1e-4])
-        else:
-            running_stats = jnp.array([0, 1, 1e-4])
-
-        trainer.store.stats = {}
-        for net_key in trainer.store.trainer_agent_net_keys.keys():
-            trainer.store.stats[net_key] = running_stats
+            trainer.store.target_running_stats_fn = compute_running_mean_var_count
+
+        trainer.store.target_stats = {}
+        for agent in trainer.store.trainer_agent_net_keys.keys():
+            trainer.store.target_stats[agent] = dict(
+                mean=jnp.array([0]), var=jnp.array([0]), count=jnp.array([1e-4])
+            )
+
+        # Initilaise observations running statisitics here
+        # We only need to modify the variance since this is actually intiailised
+        # in the training/trainer on_building_init_end callback.
+        if self.config.normalize_observations:
+            trainer.store.norm_obs_running_stats_fn = update_and_normalize_observations
+            for agent in trainer.store.trainer_agent_net_keys.keys():
+                obs_shape = len(
+                    trainer.store.obs_norm_params[constants.OBS_NORM_STATE_DICT_KEY][
```

Comment:
Maybe just create a variable as mentioned in my other comment. If there is any other place where this is also applicable please update the code.